include/seqan3/range/views/minimiser_hash.hpp

```diff
@@ -0,0 +1,148 @@
+        static_assert(semialphabet<reference_t<urng_t>>,
+            "The range parameter to views::minimiser_hash must be over elements of seqan3::semialphabet.");
+
+        if (shape.size() > window_size)
+            throw std::invalid_argument{"The size of the shape cannot be greater than the window size."};
+
+        return std::forward<urng_t>(urange) | seqan3::views::kmer_hash(shape)
+                                            | std::views::transform([seed] (int i) { return i ^ seed; })
+                                            | seqan3::views::minimiser(window_size - shape.size() + 1);
+    }
+};
+//![adaptor_def]
+
+} // namespace seqan3::detail
+
+namespace seqan3::views
+{
+
+/*!\name Alphabet related views
+ * \{
+ */
+
+/*!\brief                    Computes minimisers for a range with a given shape, window size and seed.
+ * \tparam urng_t            The type of the range being processed. See below for requirements. [template parameter is
+ *                           omitted in pipe notation]
+ * \param[in] urange         The range being processed. [parameter is omitted in pipe notation]
+ * \param[in] shape          The seqan3::shape that determines how to compute the hash value.
+ * \param[in] window_size    The window size to use.
+ * \param[in] seed           The seed to use. Default: 0x8F3F73B5CF1C9ADE.
+ * \returns                  A range of std::integral where each value is the minimiser of the resp. window.
+ *                           See below for the properties of the returned range.
+ * \ingroup views
+ *
+ * \details
+ *
+ * A sequence can be presented by a small number of k-mers (minimisers). For a given shape and window size all k-mers
+ * are determined in the forward strand and only the lexicographically smallest k-mer is saved for
+ * one window. This process is repeated over every possible window of a sequence. If consecutive windows share a
+ * minimiser, it is saved only once. It might happen that a minimiser changes only slightly when sliding
+ * the window over the sequence. For instance, when a minimiser starts with a repetition of A’s, then in the next window
+ * it is highly likely that the minimiser will start with a repetition of A’s as well. Because it is only one A shorter,
+ * depending on how long the repetition is this might go on for multiple window shifts. Saving these only slightly
+ * different minimiser makes no sense because they contain no new information about the underlying sequence plus
+ * sequences with a repetition of A’s will be seen as more similar to each other than they actually are.
+ * As [Marçais et al.](https://doi.org/10.1093/bioinformatics/btx235) have shown, randomizing the order of the k-mers
+ * can solve this problem. Therefore, a random seed is used as a default to XOR all k-mers, thereby randomzing the
+ * order. The user can change the seed to any other value he or she thinks is useful. A seed of 0 is returning the
+ * lexicographical order.
```

Comment:
Maybe you can do 2 paragraphs, one that explains the general idea and one that explains why you do the xor.
If you want to, you can also add pictures - we probably have some lying around somewhere

Reply:
I would add a picture, when minimiser_hash also handles the reverse strand? Because most of our example pictures include them. 
Plus, where would I store a picture? Like in which folder?

Reply:
> I would add a picture, when minimiser_hash also handles the reverse strand? Because most of our example pictures include them.

Ah right, then we can wait anyway.
> Plus, where would I store a picture? Like in which folder?

Probably somewhere in the `doc` folder, but we can postpone this